### PR TITLE
🌐 Improve German translation

### PR DESCRIPTION
**Pull request by daPhipz**
This is a suggestion how the German translation could be improved.
Since translations are a very subjective thing (especially grammar), feel free to request changes!

### What I changed

- Added some commas here and there. Yes, commas are a very discussion-worthy thing in the German language, but I believe this version is now mostly right.
- Made the German wording match the English original closer. See Line 93 for an example.
- Keep words English, if they are commonly used like this in German (e. g. Bug, Feature)
- Correct typos
- Emojis: Use adjectives for the "feeling" Smileys consistently (most of them were translated this way, but not all of them (e. g. line 1481 and 1513)

### CODE REVIEW

**Comment by daPhipz** [comment]
Do not merge yet, I noticed a bug when running `./compile-locales.sh`.
In order to function correctly, the `gettext` package has to be installed, which is not the case by default (at least on Ubuntu 20.04 and 20.10).

I will open a separate PR to fix this. Once it is merged, I'll run the script, update this PR and then it can be merged.

**Comment by daPhipz** [comment]
While desperately trying to generate the `.mo` file and commit the changes, I noticed that the `/locale` directory is mentioned in`.gitignore`. 😂
I am assuming that this is not intended?

**Comment by Schneegans** [comment]
That is actually intended. As the `*.mo` files are binary files which can be easily generated using the provided scripts, they should not be part of the source tree. In fact, they are automatically generated for each new release (see https://github.com/Schneegans/Fly-Pie/blob/develop/.github/workflows/deploy.yml#L21).

Is it still not working on your side? If think that this line: https://github.com/Schneegans/Fly-Pie/pull/34/files#diff-6686c8173bf4fa70e9c9755259206f2c5c4fa9fde6269d36e5262dcc09abcf25R14 might cause issues. From my limited experience it seems that the language code (in this case `de_DE`, previously only `de`) must match the code in the path to the corresponding `*.mo` file (e.g. `locale/de/LC_MESSAGES/flypie.mo`). The `compile-locales.sh` uses the filename (`de.po`) to generate the path to the `*.mo` file.

Why did you change the language code? I have not much experience in localizations but the majority of applications on my machine seem to use `de` rather than the more specific `de_DE`.

I'll look into this PR in more detail later this afternoon!

File: po/de.po
Comment (Schneegans) [inline po/de.po:14]: Did you change this on purpose?
```suggestion
"Language: de\n"
```

File: po/de.po
Comment (Schneegans) [inline po/de.po:338]: I fear that this is a bit too long for that button. As the context is clear from the button's position, maybe
```suggestion
msgstr "Zufällig erzeugen"
```

File: po/de.po
Comment (Schneegans) [inline po/de.po:662]: This is also too long. Maybe
```suggestion
msgstr "Das wird nicht von den Voreinstellungen beeinflusst."
```

File: po/de.po
Comment (Schneegans) [inline po/de.po:1513]: This is a submenu title, so I would go for the original:
```suggestion
msgstr "Überraschte Smileys"
```

File: po/de.po
Comment (Schneegans) [inline po/de.po:1565]: This is also a submenu heading. So the original should be kept:
```suggestion
msgstr "Pfeile"
```

**Review by Schneegans — CHANGES_REQUESTED** [review]
Thank you for fixing all those spelling errors! Only a few minor remarks :smile:

**Comment by daPhipz** [comment]
> As the `*.mo` files are binary files which can be easily generated using the provided scripts, they should not be part of the source tree.

That's a good point, didn't think of that.

> Why did you change the language code? I have not much experience in localizations but the majority of applications on my machine seem to use `de` rather than the more specific `de_DE`.

I used `gtranslator` to edit the translation, one of the programs you suggested. I think this setting was done by the program automatically.
AFAIK, `de` is used as fallback if no other, more specialized German locale is used - so this should be fine generally. Only if there were some special words like "Tomate" ("Paradeiser" in `de_AT`), we would need this specialization.
Should I change it back to just `de` then?

The script works now, it produces a `.mo` file - I just had to install the missing package.

I will look into your requests in the coming days, not sure when I will have the time since I am at my girlfriend's this week.

Thanks for double-checking the translation!

**Comment by Schneegans** [comment]
Then I think that we should stick to `de` until someone comes up with a more specialized localization!

**Comment by daPhipz** [comment]
There you go! I implemented your corrections from above and changed the locale back to `de`.

**Review by Schneegans — APPROVED** [review]

**Pull request merged**
